improve html templates

slomo opened the pull request:
accidentally screwed up #34, this is the right one now

slomo committed:
replace buttons that can be links with links

slomo committed:
remove capitalized html tags

slomo committed:
add djlint for html templates

slomo committed:
fix existing tempalte formating

Commit:
[LINTING] Remove all linting errors

Motivation
  - quality assurance should pass without any errors
  - [CSS] Back to old style; using button, not href

How:
  - removed the following errors:

```
src/templates/base_template.html
─────────────────────────────────────────────────────────────────────────────────────────────
H005 2:0 Html tag should have lang attribute. <html> <head>
H031 2:0 Consider adding meta keywords. <html> <head>

src/templates/index.html
─────────────────────────────────────────────────────────────────────────────────────────────
H012 14:4 There should be no spaces around attribute =. <form action="" meth
H012 16:8 There should be no spaces around attribute =. <input id="name" typ
H012 17:8 There should be no spaces around attribute =. <input type = "submi

src/templates/player_info.html
─────────────────────────────────────────────────────────────────────────────────────────────
H025 50:4 Tag seems to be an orphan. </ul>

src/templates/task_basic.html
─────────────────────────────────────────────────────────────────────────────────────────────
H020 7:4 Empty tag pair found. Consider removing. <p></p>
H012 10:4 There should be no spaces around attribute =. <form action="" meth
H012 12:8 There should be no spaces around attribute =. <input type = "text"
```

Nimuedaertya commented:
Layout and CSS a little bit changed. Would like to keep the old format. First image shows, should-be-state and second is the how-it-is-state.

![Wed 11 Sep 13:10:44 CEST 2024](https://github.com/user-attachments/assets/2a5a7ba8-0a5b-4c76-9884-9c8fbf80b737)
![Wed 11 Sep 13:04:59 CEST 2024](https://github.com/user-attachments/assets/ae2a1a1e-2363-45b6-a535-cf53774c4869)

Nimuedaertya reviewed (approved):
I like the linting for the jinja files very much.
For changing the layout from buttons to href, we need some proper styling. Hence I reverted the changes for now - sorry :D

slomo committed:
Revert "[LINTING] Remove all linting errors"

This reverts commit 78e93860af04c9ffa82d1bf349b0ddf0f147a53c.

slomo committed:
fix button and align css

 - fixes issues with styling links as buttons
 - remove all unneeded styles from css files
 - remove unused (an non working) tailwind imports
 - order styles according to page they belong to
 - make css class naming consistently snake case

slomo committed:
make djlint happy

slomo committed:
change text alignment for readability

 - paragraphs are longer texts, those should be not aligned center for
   to improve readability

 - see: https://practicaltypography.com/centered-text.html)

slomo committed:
simplyfy css pipeline by removing it

 - no functionality of Flask-Asset was really used
 - bundling for that size of css doesn't make sense and makes it more
   complicated to develop css

slomo committed:
Merge remote-tracking branch 'origin/main' into feature/improve_html_templates
